geode-junit/src/test/java/org/apache/geode/test/junit/rules/ConcurrencyRuleTest.java

```diff
@@ -418,6 +419,31 @@ public void timeoutValueIsRespected(Execution execution) {
     });
   }
 
+  @Test
+  public void afterFailsIfThreadsWereNotRun() {
+    Callable<Integer> c1 = () -> {
+      return 2;
+    };
+
+    Callable<String> c2 = () -> {
+      return "some string";
+    };
+
+    concurrencyRule.add(c1).expectValue(2);
+    concurrencyRule.add(c1).expectValue(2).repeatForIterations(5);
+    concurrencyRule.executeInParallel();
+
+    concurrencyRule.add(c1).expectValue(3);
+    concurrencyRule.add(c2).expectValue("some string");
+    try {
+      concurrencyRule.after();
+      fail("exception should have been thrown");
```

Comment:
Rather than `fail()` aasertions, use assertj's `assertThatThrownBy()`.

There's a current discussion thread on the dev@geode.apache.org regarding assertions in tests, "[DISCUSS] Which assert is the right one to use moving forward?"

Reply:
👍 will do